src/main/java/org/apache/maven/doxia/DefaultConverter.java

```diff
@@ -78,57 +77,160 @@
 
-    private static final String CONFLUENCE_SINK = "confluence";
+        DoxiaFormat( String roleHint, String extension, boolean hasParser, boolean hasSink )
+        {
+            this( roleHint, extension, null, hasParser, hasSink );
+        }
 
-    private static final String DOCBOOK_SINK = "docbook";
+        DoxiaFormat( String roleHint, String extension, String firstElement, boolean hasParser, boolean hasSink )
+        {
+            this.roleHint = roleHint;
+            this.extension = extension;
+            this.firstElement = firstElement;
+            this.hasParser = hasParser;
+            this.hasSink = hasSink;
+        }
 
-    private static final String FO_SINK = "fo";
+        /**
+         * 
+         * @return the primary extension used with this format
+         */
+        public String getExtension()
+        {
+            return extension;
+        }
 
-    private static final String ITEXT_SINK = "itext";
+        public boolean hasParser()
+        {
+            return hasParser;
+        }
 
-    private static final String LATEX_SINK = "latex";
+        public boolean hasSink()
+        {
+            return hasSink;
+        }
 
-    private static final String RTF_SINK = "rtf";
+        /**
+         * 
+         * @return {@code true} in case this format is XML based
+         */
+        public boolean isXml()
+        {
+            return firstElement != null;
+        }
 
-    private static final String TWIKI_SINK = "twiki";
+        /**
+         * @param plexus not null
+         * @return an instance of <code>Parser</code> depending on the format.
+         * @throws ComponentLookupException if could not find the Parser for the given format.
+         * @throws IllegalArgumentException if any parameter is null
+         */
+        public Parser getParser( PlexusContainer plexus )
+            throws ComponentLookupException
+        {
+            if ( !hasParser )
+            {
+                throw new IllegalStateException( "The format " + this.toString() + " is not supported as parser!" );
+            }
+            Objects.requireNonNull( plexus, "plexus is required" );
 
-    private static final String XDOC_SINK = "xdoc";
+            return (Parser) plexus.lookup( Parser.ROLE, roleHint );
+        }
 
-    private static final String XHTML_SINK = "xhtml";
+        /**
+         * @param plexus not null
+         * @return an instance of <code>SinkFactory</code> depending on the given format.
+         * @throws ComponentLookupException if could not find the SinkFactory for the given format.
+         * @throws IllegalArgumentException if any parameter is null
+         */
+        public SinkFactory getSinkFactory( PlexusContainer plexus )
+            throws ComponentLookupException
+        {
+            if ( !hasSink )
+            {
+                throw new IllegalStateException( "The format " + this.toString() + " is not supported as sink!" );
```

Comment:
same here

Reply:
Fixed in https://github.com/apache/maven-doxia-converter/pull/38/commits/4fa08336ee9db0ae4d4eb4c22b68f8ebb22f6d7d.